internal/ast/ast.go

```diff
@@ -273,25 +273,59 @@ type (
 		Cond Expr       // While loop condition.
 		Body *BlockStmt // Body of while.
 	}
+
+	// SwitchStatement node represents a switch statement
+	SwitchStatement struct {
+		Position
+		Value Expr       // Value to switch case
+		Body  *BlockStmt // Switch body
+	}
+
+	// SwitchCase node represents a single switch case
+	SwitchCase struct {
+		Position
+		Value Expr   // Condition to check to enter this case
```

Comment:
What do you think rename this field to `Cond`? Like on `IfStmt`

Reply:
Done!